### PR TITLE
More complex authentication

### PR DESCRIPTION
**Pull request by Juhi0326**

### CODE REVIEW

File: src/app/auth.service.ts [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+  SignIn(email, password) {
+    return this.afAuth.auth.signInWithEmailAndPassword(email, password)
+      .then((result) => {
+        this.ngZone.run(() => {
+
+          this.router.navigate(['customers']);
+        });
+        this.SetUserData(result.user);
+      }).catch((error) => {
+        window.alert(error.message);
```
Comment (dbreuer) [inline src/app/auth.service.ts]: Use global or module error handler and not DOM window API.

File: src/app/auth.service.ts [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+  SignUp(email, password) {
+    return this.afAuth.auth.createUserWithEmailAndPassword(email, password)
+      .then((result) => {
+        /* Call the SendVerificaitonMail() function when new user sign
+        up and returns promise */
+
+        this.SendVerificationMail();
+        this.SetUserData(result.user);
+      }).catch((error) => {
+        window.alert(error.message);
```
Comment (dbreuer) [inline src/app/auth.service.ts]: Error handler

File: src/app/secure-inner-pages.guard.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    public authService: AuthService,
+    public router: Router
+  ) { }
+
+  canActivate(
+    next: ActivatedRouteSnapshot,
+    state: RouterStateSnapshot): Observable<boolean> | Promise<boolean> | boolean {
+    if (this.authService.isLoggedIn) {
+       window.alert('Már be vagy lépve!');
+       this.router.navigate(['customers']);
```
Comment (dbreuer) [inline src/app/secure-inner-pages.guard.ts]: Missing return false;

File: src/app/users/User.ts [hunk 4]
```diff
@@ -1,5 +1,8 @@
 export interface User {
-  userName: String;
-  password?: String;
-  id?: Number;
+  uid: string;
+  email: string;
+  displayName: string;
+  photoURL: string;
```
Comment (dbreuer) [inline src/app/users/User.ts]: Is the photoURL mandatory, it should be optional property.

File: src/app/users/forgot-password/forgot-password.component.html [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+      </div>
+
+      <!-- Calling ForgotPassword from AuthService Api -->
+      <div class="formGroup">
+        <input type="submit" class="btn btnPrimary nagyit2" value="Reset Password" (click)="authService.ForgotPassword(passwordResetEmail.value)">
+      </div>
+    </div>
+
+    <div class="redirectToLogin">
+      <span>Go back to ? <span class="redirect nagyit2" (click)="logIn()" >Log In</span></span>
```
Comment (dbreuer) [inline src/app/users/forgot-password/forgot-password.component.html]: please do not mix Hungarian names with English for class

File: src/app/users/forgot-password/forgot-password.component.ts [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+  styleUrls: ['./forgot-password.component.css']
+})
+export class ForgotPasswordComponent implements OnInit {
+
+  constructor(
+    public authService: AuthService,
+    private router: Router
+  ) { }
+
+  ngOnInit() {
```
Comment (dbreuer) [inline src/app/users/forgot-password/forgot-password.component.ts]: Just remove if you not using it.

File: src/app/users/forgot-password/forgot-password.component.html (hunk 5 above)
Comment (dbreuer) [inline src/app/users/forgot-password/forgot-password.component.html]: If there is no complex functionality behind the login than its enough to use routerLink here and remove the function.

File: src/app/users/sign-in/sign-in.component.html [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+        <br>
+      </div>
+
+      <div class="formGroup row">
+        <input type="password" class="col-sm-6 col-form-label" placeholder="Password" #userPassword required>
+      </div>
+
+      <!-- Calling SignIn Api from AuthService -->
+      <div class="formGroup row">
+        <input type="button" class="btn btnPrimary nagyit2" value="Log in" (click)="authService.SignIn(userName.value, userPassword.value)">
```
Comment (dbreuer) [inline src/app/users/sign-in/sign-in.component.html]: Please update the Value based on the **Log in** not same as **Sign In**

File: src/app/users/sign-in/sign-in.component.html [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+      </div>
+
+      <div class="formGroup row">
+        <span class="or"><span class="orInner">Or</span></span>
```
Comment (dbreuer) [inline src/app/users/sign-in/sign-in.component.html]: Dont use camelCase names for class please. **or-inner**

File: src/app/users/sign-in/sign-in.component.ts [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+  constructor(
+    public authService: AuthService,
+    private router: Router
+  ) { }
+
+  ngOnInit() {
+  }
+
+  signUp() {
+    this.router.navigate(['login/register-user']);
```
Comment (dbreuer) [inline src/app/users/sign-in/sign-in.component.ts]: These could be removed and just use the routerLink directive.

File: src/app/users/verify-email/verify-email.component.html [hunk 10]
```diff
@@ -0,0 +1,29 @@
+<div class="displayTable">
+  <div class="displayTableCell">
+
```
Comment (dbreuer) [inline src/app/users/verify-email/verify-email.component.html]: Remove white spaces (clean code)

File: src/app/auth.service.ts [hunk 11]
```diff
@@ -0,0 +1,136 @@
...
+      });
+  }
+
+  // Send email verfificaiton when new user sign up
+  SendVerificationMail() {
```
Comment (dbreuer) [inline src/app/auth.service.ts]: camelCase function names mean start with a lowercase.

**Comment by Juhi0326** [comment]
Ok, thanks, i will check these comments on friday morning!

David Breuer <notifications@github.com> ezt írta (időpont: 2019. jún. 26.,
Sze 18:19):

File: src/app/users/forgot-password/forgot-password.component.ts (hunk 6 above)
Comment (Juhi0326) [reply src/app/users/forgot-password/forgot-password.component.ts]: done

File: src/app/users/User.ts (hunk 4 above)
Comment (Juhi0326) [reply src/app/users/User.ts]: done

File: src/app/users/verify-email/verify-email.component.html (hunk 10 above)
Comment (Juhi0326) [reply src/app/users/verify-email/verify-email.component.html]: done

File: src/app/secure-inner-pages.guard.ts (hunk 3 above)
Comment (Juhi0326) [reply src/app/secure-inner-pages.guard.ts]: done

File: src/app/auth.service.ts (hunk 11 above)
Comment (Juhi0326) [reply src/app/auth.service.ts]: done

File: src/app/users/sign-in/sign-in.component.html (hunk 8 above)
Comment (Juhi0326) [reply src/app/users/sign-in/sign-in.component.html]: done

File: src/app/users/forgot-password/forgot-password.component.html (hunk 5 above)
Comment (Juhi0326) [reply src/app/users/forgot-password/forgot-password.component.html]: done

File: src/app/auth.service.ts (hunk 1 above)
Comment (Juhi0326) [reply src/app/auth.service.ts]: ok, i have created a messages service for this alerts.

File: src/app/auth.service.ts (hunk 2 above)
Comment (Juhi0326) [reply src/app/auth.service.ts]: ok, i have created a messages service for this alerts.

File: src/app/users/forgot-password/forgot-password.component.html (hunk 5 above)
Comment (Juhi0326) [reply src/app/users/forgot-password/forgot-password.component.html]: done

File: src/app/users/sign-in/sign-in.component.html (hunk 7 above)
Comment (Juhi0326) [reply src/app/users/sign-in/sign-in.component.html]: done

File: src/app/users/sign-in/sign-in.component.ts (hunk 9 above)
Comment (Juhi0326) [reply src/app/users/sign-in/sign-in.component.ts]: done

**Comment by Juhi0326** [comment]
I have fixed all problems

**Pull request merged**